src/engine/loaders/FileReader.js

```diff
@@ -88,6 +88,14 @@ class FileReader extends React.Component {
 
     if (vol.m_dataArray !== null) {
       console.log(`success loaded volume from ${fileNameIn}`);
+
+      const RECENT_FILES_KEY = 'recentFiles';
+      const arrRecentFiles = JSON.parse(localStorage.getItem(RECENT_FILES_KEY)) || [];
+
+      const limitedRecentFiles = arrRecentFiles.slice(0, 2);
+
+      localStorage.setItem(RECENT_FILES_KEY, JSON.stringify([{ fileName: fileNameIn, timestamp: Date.now() }, ...limitedRecentFiles]));
```

Comment:
Each time the callback is being executed twice, please, help me with investigation here 🙏🏼

Reply:
The setItem is only executed once, but the callback which it is in is executed twice for each file opening.